samza-sql-shell/src/main/java/org/apache/samza/sql/client/cli/CliEnvironment.java

```diff
@@ -173,11 +187,21 @@ public SqlExecutor getExecutor() {
     return executor;
   }
 
+  public List<CommandHandler> getCommandHandlers() { return commandHandlers; }
+
   private void createShellExecutor(String executorClassName) throws ExecutorException {
+    executor = (SqlExecutor) createInstance(executorClassName);
+  }
+
+  private void createCommandHandler(String handlerClassName) throws ExecutorException {
+    commandHandlers.add((CommandHandler) createInstance(handlerClassName));
+  }
+
+  private Object createInstance(String className) throws ExecutorException {
```

Comment:
It's fine to throw a `ExecutorException` for a method of `createShellExecutor`, but would be misleading for a method named `createInstance` as it creates `commandhandler`s as well.

Reply:
makes sense, I added CommandHandlerException class, CommandHandler now throws it, and createInstance here throws Exception, so that the createShellExecutor throws ExecutorException while createCommandHandler throws CommandHandlerException